src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,99 @@
+package duke;
+import java.io.File;
+import java.io.FileWriter;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Encapsulates a store that stores a file containing the list of tasks.
+ */
+public class Storage {
+    private File file;
+
+    /**
+     * Creates a new Storage object.
+     * @param filepath The path of the file to be created to store the list.
+     */
+    Storage(String filepath) {
+        this.file = new File(filepath);
+    }
+
+    /**
+     * Stores the current TaskList into a file
+     * @param list The TaskList object containing an ArrayList of Tasks
+     * @throws DukeException Throws a DukeException
+     */
+    public void save(TaskList list) throws DukeException {
+        ArrayList<String> temp = new ArrayList<>();
+        try {
+            FileWriter fw = new FileWriter(file);
+            for (int i = 1; i <= list.numberOfTasks(); i++) {
+                temp.add(list.getTask(i).sendOutputToFile());
+            }
+            fw.write(String.join("\n", temp));
+            fw.close();
+        } catch (Exception e) {
+            throw new DukeException("Error when adding file");
+        }
+    }
+
+    /**
+     * Loads Tasks from the saved text file to return the ArrayList of Tasks
+     * @return The ArrayList of tasks from the peviously saved text file
+     * @throws DukeException Throws a DukeException
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            if (!file.exists()) {
+                file.getParentFile().mkdir();
+            }
+            if (file.exists()) {
```

Comment:
I think there is slightly too many levels of nesting here and may affect readability. Perhaps you can restructure the conditionals here to have your `while` loop extracted out of the nesting? I feel that would also make the 'happy path' more prominent!